packages/sanity/src/desk/panes/document/DocumentPaneProvider.tsx

```diff
@@ -206,6 +210,11 @@ export const DocumentPaneProvider = memo((props: DocumentPaneProviderProps) => {
     ? sinceAttributes
     : editState?.published || null
 
+  const fieldActions: DocumentFieldAction[] = useMemo(
+    () => (schemaType ? fieldActionsResolver({documentId, documentType, schemaType}) : []),
+    [documentId, documentType, fieldActionsResolver, schemaType]
```

Comment:
Each hook is rendered in a separate component, so it will work even if number of actions change => for each component, the number of hooks will be stable.